src/flask_cognito_lib/utils.py

```diff
@@ -33,3 +33,29 @@ class CognitoTokenResponse:
     refresh_token: Optional[str] = None
     id_token: Optional[str] = None
     error: Optional[str] = None
+
+
+@dataclass
+class CognitoExtraAuthorizeParams:
```

Comment:
Whilst I appreciate that a dataclass does offer a little bit more type safety, I'd prefer if we simplified this a bit and add a new configuration option for this instead. Please could you add a new configuration parameter, that is an `Optional[Dict[str, str]]`, defaulting to `None`, named something like `AWS_COGNITO_AUTH_EXTRA_ARGS`. 

This means:
* We automatically will support any params that Cognito may add in the future (or remove), without needing to maintain a copy in this package, or make new releases whenever AWS make changes to Cognito 
* We can remove the need to make any changes made to the decorators, since in the Cognito service we can just pull the args out of the configuration, if they are set

Additionally, in the `exchange_code_for_token` function in the Cognito service, we should warn the user if they have set extra args that this library already uses (e.g. we don't want a user to override things like `code_challenge`, `nonce` etc.) that said parameters will be ignored.

Mainly I think both reduces the future maintenance burden, and keeps the interface closer to how the rest of the package works - i.e. any existing users who wanted to pass say `language=es` can simply set this via configuration rather than having to update their `@cognito_login` route.